classes/chat/ChatManager.php

```diff
@@ -78,33 +78,75 @@ private function fetchPosts(?int $starting_post_id = null, int $max_posts = self
+        foreach ($result as $row) {
+            // If user_id is set (newer post) add to list for batch
+            if ($row['user_id'] > 0 && !in_array($row['user_id'], $user_ids)) {
+                $user_ids[] = $row['user_id'];
+            }
+            // If user_id is not set (older post) get data for that user based on user_name
+            else {
+                $user_data_result = $this->system->db->query("
+                    SELECT `users`.`user_id`, `users`.`staff_level`, `users`.`premium_credits_purchased`, `users`.`chat_effect`, `users`.`avatar_link`,
+                    `user_settings`.`avatar_style`, `user_settings`.`avatar_frame`
+                    FROM `users`
+                    LEFT JOIN `user_settings` ON `users`.`user_id` = `user_settings`.`user_id`
+                    WHERE `users`.`user_name` = '{$this->system->db->clean($row['user_name'])}'
+                ");
+                $user_data_result = $this->system->db->fetch($user_data_result);
+                // only add data to array if user match found
+                if ($this->system->db->last_num_rows > 0) {
+                    // if custom setting is not set
+                    if (!isset($user_data_result['avatar_style'])) {
+                        $user_data_result['avatar_style'] = "avy_round";
+                    }
+                    // if custom setting is not set
+                    if (!isset($user_data_result['avatar_frame'])) {
+                        $user_data_result['avatar_frame'] = "avy_frame_default";
+                    }
+                    $user_data_by_name[$row['user_name']] = $user_data_result;
+                }
+            }
+        }
+        // Get batch user data
+        $user_data_result = $this->system->db->query("
+            SELECT `users`.`user_id`, `users`.`staff_level`, `users`.`premium_credits_purchased`, `users`.`chat_effect`, `users`.`avatar_link`,
+            `user_settings`.`avatar_style`, `user_settings`.`avatar_frame`
+            FROM `users`
+            LEFT JOIN `user_settings` ON `users`.`user_id` = `user_settings`.`user_id`
+            WHERE `users`.`user_id` IN ('" . implode(', ', $user_ids) . "')
+        ");
+        $user_data_result = $this->system->db->fetch_all($user_data_result);
+        foreach ($user_data_result as $row) {
+            // if custom setting is not set
+            if (!isset($row['avatar_style'])) {
+                $row['avatar_style'] = "avy_round";
+            }
+            // if custom setting is not set
+            if (!isset($row['avatar_frame'])) {
+                $row['avatar_frame'] = "avy_frame_default";
+            }
+            $user_data_by_id[$row['user_id']] = $row;
+        }
         $posts = [];
-        while($row = $this->system->db->fetch($result)) {
+        foreach ($result as $row) {
             $post = ChatPostDto::fromDb($row);
 
             //Base data
             $post->avatar = './images/default_avatar.png';
```

Comment:
Just a note for us, could move defaults like these into the ChatPostDto.

Reply:
Good point, taking note as well